lib/blockchain/chain.js

```diff
@@ -659,17 +659,18 @@ class Chain extends AsyncEmitter {
     const hardened = state.hasHardening();
     const view = new CoinView();
     const height = prev.height + 1;
-    const cb = block.txs[0];
-
-    assert(view.bits.spend(this.db.field, cb));
 
-    view.addTX(cb, height);
+    assert(block.treeRoot.equals(this.db.treeRoot()));
 
-    for (let i = 1; i < block.txs.length; i++) {
+    for (let i = 0; i < block.txs.length; i++) {
       const tx = block.txs[i];
 
-      assert(await view.spendInputs(this.db, tx),
-        'BUG: Spent inputs in historical data!');
+      if (i === 0) {
+        assert(view.bits.spend(this.db.field, tx));
+      } else {
+        assert(await view.spendInputs(this.db, tx),
+          'BUG: Spent inputs in historical data!');
+      }
```

Comment:
Besides the tree root assertion, this change looks like a noop to me -- what was fixed?